### PR TITLE
Hash pick/omit renamed to select/reject

### PR DESCRIPTION
**Pull request by sdogruyol**
This is continuation of https://github.com/manastech/crystal/pull/1183 and renamed version.

### CODE REVIEW

File: src/hash.cr [hunk 1]
```diff
@@ -470,6 +470,48 @@ class Hash(K, V)
...
+  # Removes every element except the given ones.
+  #
+  # ```
+  # h = {"a": 1, "b": 2, "c": 3, "d": 4}.select!("a", "c")
+  # h #=> {"a": 1, "c": 3}
+  # ```
+  def select!(*keys)
+    each { |k, v| delete(k) unless keys.includes?(k) }
+    self
+
```
Comment (jhass) [inline src/hash.cr]: Missing `end`

File: spec/std/hash_spec.cr [hunk 2]
```diff
@@ -530,6 +530,48 @@ describe "Hash" do
...
+      h2 = h.select(:b, :c)
+      h.should eq({ a: 3, b: 6, c: 9 })
+    end
+  end
+
+  describe "select!" do
+    assert { { a: 2, b: 3 }.select!(:b, :d).should eq({ b: 3 }) }
+    assert { { a: 2, b: 3 }.select!.should eq({ } of Symbol => Int32) }
+    assert { { a: 2, b: 3 }.select!(:b, :a).should eq({ a: 2, b: 3 }) }
+    it "does not change currrent hash" do
```
Comment (luislavena) [inline spec/std/hash_spec.cr]: Nitpick: It **does** change current hash.

Comment (sdogruyol) [inline spec/std/hash_spec.cr]: Oops

**Comment by jhass** [comment]
Thank you!

**Pull request merged**